flycheck.el

```diff
@@ -6924,7 +6924,16 @@ See URL `http://www.erlang.org/'."
 
 (defun flycheck-rebar3-project-root (&optional _checker)
   "Return directory where =rebar.config= is located."
-  (locate-dominating-file buffer-file-name "rebar.config"))
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
```

Comment:
@joedevivo Mmh, I'm not sure whether that's not too eager: Is there no reason why anyone would ever nest two proper rebar projects under a top-level project?   Would you never use "nested" projects with rebar?